sdk/extensions/Azure.Extensions.AspNetCore.DataProtection.Keys/tests/AzureDataProtectionBuilderExtensionsTests.AzureKeyVaultXmlEncryptor.cs

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Identity;
+using Azure.Security.KeyVault.Keys.Cryptography;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.DataProtection.KeyManagement;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using NUnit.Framework;
+
+namespace Azure.Extensions.AspNetCore.DataProtection.Keys.Tests
+{
+    public partial class AzureDataProtectionBuilderExtensionsTests
+    {
+        [Test]
+        public void ProtectKeysWithAzureKeyVault_With_String_And_Credential_Uses_AzureKeyVaultXmlEncryptor()
+        {
+            // Arrange
+            var serviceCollection = new ServiceCollection();
+            var builder = serviceCollection.AddDataProtection();
+
+            // Act
+            builder.ProtectKeysWithAzureKeyVault("http://www.example.com/dummyKey", new DefaultAzureCredential());
+            var services = serviceCollection.BuildServiceProvider();
+
+            // Assert
+            var options = services.GetRequiredService<IOptions<KeyManagementOptions>>();
+            Assert.IsInstanceOf<AzureKeyVaultXmlEncryptor>(options.Value.XmlEncryptor);
+        }
+
+        [Test]
+        public void ProtectKeysWithAzureKeyVault_With_Uri_And_Credential_Uses_AzureKeyVaultXmlEncryptor()
+        {
+            // Arrange
+            var serviceCollection = new ServiceCollection();
+            var builder = serviceCollection.AddDataProtection();
+
+            // Act
+            builder.ProtectKeysWithAzureKeyVault(new Uri("http://www.example.com/dummyKey"), new DefaultAzureCredential());
```

Comment:
```suggestion
            builder.ProtectKeysWithAzureKeyVault(new Uri("http://www.example.com/dummyKey"), new MockCredential());
```

As these are all non-live tests, I think we'll want to ensure they're using the mock and not a credential with ties to live Azure resources.